File: go/1/carro.go
package main

import "fmt"

type Carro struct {
	marca     string
	modelo    string
	ano       int
	velocidade int
}

func (c *Carro) Acelerar(incremento int) {
	if incremento > 0 {
		c.velocidade += incremento
		fmt.Printf("O carro acelerou %d km/h.\n", incremento)
	} else {
		fmt.Println("O valor de aceleração deve ser positivo.")
	}
}

func (c *Carro) Frear(decremento int) {
	if decremento > 0 && c.velocidade-decremento >= 0 {
		c.velocidade -= decremento
		fmt.Printf("O carro freou %d km/h.\n", decremento)
	} else {
		fmt.Println("A frenagem não pode resultar em velocidade negativa.")
	}
}

func (c *Carro) ExibirVelocidade() {
	fmt.Printf("Velocidade atual: %d km/h\n", c.velocidade)
}

func (c *Carro) ExibirDetalhes() {
	fmt.Printf("Marca: %s\n", c.marca)
	fmt.Printf("Modelo: %s\n", c.modelo)
	fmt.Printf("Ano: %d\n", c.ano)
}

File: python/1/Carro.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, incremento):
        if incremento > 0:
            self.velocidade += incremento
            print(f"O carro acelerou {incremento} km/h.")
        else:
            print("O valor de aceleração deve ser positivo.")

    def frear(self, decremento):
        if decremento > 0 and self.velocidade - decremento >= 0:
            self.velocidade -= decremento
            print(f"O carro freou {decremento} km/h.")
        else:
            print("A frenagem não pode resultar em velocidade negativa.")

    def exibir_velocidade(self):
        print(f"Velocidade atual: {self.velocidade} km/h")

    def exibir_detalhes(self):
        print(f"Marca: {self.marca}")
        print(f"Modelo: {self.modelo}")
        print(f"Ano: {self.ano}")
